src/solvers/td3.ml

```diff
@@ -627,22 +629,40 @@ module WP =
             )
         in
 
+        let force_reanalyze = StringSet.of_list @@ GobConfig.get_string_list "incremental.force-reanalyze.funs" in
+        let reluctant = GobConfig.get_bool "incremental.reluctant.on" in
+        let reanalyze_entry f =
+          (* destabilize the entry points of a changed function when reluctant is off,
+             or the function is to be force-reanalyzed  *)
+          (not reluctant) || StringSet.mem f.svar.vname force_reanalyze
+        in
+
         let obsolete_ret = HM.create 103 in
         let obsolete_entry = HM.create 103 in
         let obsolete_prim = HM.create 103 in
+
+        (* When reluctant is on:
+           Only add function entry nodes to obsolete_entry if they are in force-reanalyze *)
         List.iter (fun f ->
-            mark_node obsolete_entry f (FunctionEntry f);
-            mark_node obsolete_ret f (Function f);
+            if reanalyze_entry f then
+              (* collect function entry for eager destabilization *)
+              mark_node obsolete_entry f (FunctionEntry f)
+            else
+              (* collect function return for reluctant analysis *)
+              mark_node obsolete_ret f (Function f)
```

Comment:
Wasn't there some reason before, why `Function` was marked obsolete even though obsoleting/destabilizing `FunctionEntry` should do the same?

Reply:
The unknowns collected in `obsolete_ret` are really only used in the `reluctant` mode, to collect `old_ret`, a mapping of unknowns to their old value in `rho` and their `infl` set. This `old_ret` is then used iterate over and call `solve` on; if the new result is different than the old one for an unknown, it is destabilized.
https://github.com/goblint/analyzer/blob/b6b9b44909e9709086af38f88ad799afe0c06db7/src/solvers/td3.ml#L645-L653
So in the regular incremental mode, only `obsolete_entry`  and `obsolete_prim` are used for eager destabilization.

Reply:
Ah, right, yes. I missed that they get collected into different sets and just remembered some old discussion.